openfl/component/aggregator/aggregator.py

```diff
@@ -177,6 +178,13 @@ def __init__(
 
         self.collaborator_task_weight = {}  # {TaskResultKey: data_size}
 
+        # maintain a list of collaborators that have completed task and
+        # reported results in a given round
+        self.collaborators_done = []
+
+        # Initialize a mutex lock for thread safety
+        self.mutex = Lock()
```

Comment:
nit - a mutex is a lower-level construct (used for the implementation of a Lock under the hood). Therefore, I suggest to simply write here:
```python
self.lock = Lock()
```

Reply:
Updated as per your suggestion. Thanks!